File: MainApp/src/main/java/com/sleticalboy/learning/base/BaseRVHolder.java
package com.sleticalboy.learning.base;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.LayoutRes;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

/**
 * Created on 20-8-23.
 *
 * @author binlee sleticalboy@gmail.com
 */
public abstract class BaseRVHolder<DATA> extends RecyclerView.ViewHolder {

    public BaseRVHolder(@NonNull final ViewGroup parent, @LayoutRes int layout) {
        super(LayoutInflater.from(parent.getContext()).inflate(layout, parent, false));
    }

    public abstract void bindData(@NonNull DATA data, int position);
}

File: MainApp/src/main/java/com/sleticalboy/learning/base/BaseRVHolder.kt
package com.sleticalboy.learning.base

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.recyclerview.widget.RecyclerView


/**
 * Created on 20-8-23.
 *
 * @author binlee sleticalboy@gmail.com
 */
abstract class BaseRVHolder<DATA>(parent: ViewGroup, @LayoutRes layout: Int) : RecyclerView.ViewHolder(LayoutInflater.from(parent.context).inflate(layout, parent, false)) {
    abstract fun bindData(data: DATA, position: Int)
}